clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java

```diff
@@ -0,0 +1,181 @@
+   */
+  public GravitinoVersion getVersion() {
+    VersionResponse resp =
+        restClient.get(
+            "api/version",
+            VersionResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.restErrorHandler());
+    resp.validate();
+
+    return new GravitinoVersion(resp.getVersion());
+  }
+
+  /** Closes the GravitinoClient and releases any underlying resources. */
+  @Override
+  public void close() {
+    if (restClient != null) {
+      try {
+        restClient.close();
+      } catch (Exception e) {
+        // Swallow the exception
+        LOG.warn("Failed to close the HTTP REST client", e);
+      }
+    }
+  }
+
+  /** Builder class for constructing a GravitinoClient. */
+  public static class Builder<T> {
+    /** The base URI for the Gravitino API. */
+    protected String uri;
+    /** The name of the metalake that the client is working on. */
+    protected String metalakeName;
+    /** The authentication provider. */
+    protected AuthDataProvider authDataProvider;
+
+    /**
+     * The constructor for the Builder class.
+     *
+     * @param uri The base URI for the Gravitino API.
+     */
+    protected Builder(String uri) {
+      this.uri = uri;
+    }
+
+    /**
+     * Sets the simple mode authentication for Gravitino
+     *
+     * @return This Builder instance for method chaining.
+     */
+    public Builder<T> withSimpleAuth() {
+      this.authDataProvider = new SimpleTokenProvider();
+      return this;
+    }
+
+    /**
+     * Sets OAuth2TokenProvider for Gravitino.
+     *
+     * @param dataProvider The OAuth2TokenProvider used as the provider of authentication data for
+     *     Gravitino Client.
+     * @return This Builder instance for method chaining.
+     */
+    public Builder<T> withOAuth(OAuth2TokenProvider dataProvider) {
+      this.authDataProvider = dataProvider;
+      return this;
+    }
+
+    /**
+     * Sets KerberosTokenProvider for the Gravitino.
+     *
+     * @param dataProvider The KerberosTokenProvider used as the provider of authentication data for
+     *     Gravitino Client.
+     * @return This Builder instance for method chaining.
+     */
+    public Builder<T> withKerberosAuth(KerberosTokenProvider dataProvider) {
+      try {
+        if (uri != null) {
+          dataProvider.setHost(new URI(uri).getHost());
+        }
+      } catch (URISyntaxException ue) {
+        throw new IllegalArgumentException("URI has the wrong format", ue);
+      }
+      this.authDataProvider = dataProvider;
+      return this;
+    }
+
+    /**
+     * Optional, set the metalake name for this client.
+     *
+     * @param metalakeName The name of the metalake that the client is working on.
+     * @return This Builder instance for method chaining.
+     */
+    public Builder<T> withMetalake(String metalakeName) {
+      this.metalakeName = metalakeName;
+      return this;
+    }
```

Comment:
This method seems will only be used by `GravitinoClient`, I think we can move this method to `ClientBuilder`.

Reply:
Ok, I moved it to child class "ClientBuilder"; the signature of build() method also changed.Please review.